contrib/mesos/pkg/service/endpoints_controller.go

```diff
@@ -325,7 +324,6 @@ func (e *endpointController) syncService(key string) {
 			containerPortAnnotations[fmt.Sprintf(meta.ContainerPortKeyFormat, portProto, pod.Status.HostIP, portNum)] = strconv.Itoa(containerPort)
 		}
 	}
-	subsets = endpoints.RepackSubsets(subsets)
```

Comment:
Strongly urge caution here. Deviating from upstream could have dire consequences later. In fact, I think we should more carefully examine the assumptions made in the validation code before attacking things here.